paasta_tools/cli/cmds/local_run.py

```diff
@@ -653,6 +678,89 @@ def check_if_port_free(port):
     return True
 
 
+def assume_aws_role(
+    instance_config, service, assume_role, assume_pod_identity, aws_profile
+):
+    """Runs AWS cli to assume into the correct role, then extract and return the ENV variables from that session"""
+    pod_identity = instance_config.get_iam_role()
+    if assume_role:
+        pod_identity = assume_role
+    if assume_pod_identity and not pod_identity:
+        print(
+            "Error: --assume-pod-identity passed but no pod identity was found for this instance",
+            file=sys.stderr,
+        )
+        sys.exit(1)
+    default_profile = ""
+    if not aws_profile:
+        with open("/nail/etc/ecosystem") as ecosystem_file:
+            default_profile = ecosystem_file.read()
+    profile = aws_profile or default_profile
+    if pod_identity and assume_pod_identity:
+        print(
+            "Attempting to assume role {} using profile {}".format(
+                pod_identity, profile
+            )
+        )
+        if os.getuid() == 0:
```

Comment:
just curious: when does this happen? is this due to the re-exec as root that we do in certain cases?

Reply:
I was generally testing with `--pull` which will re-exec as root as you mention